### DISCUSSION TITLE
MD frames

### DISCUSSION THREAD

**Discussion by lnrdwsh1ngt0n** [BradyAJohnston/MolecularNodes#684] [Q&A]
Hello everyone,
I was hoping to get some help with an issue I'm encountering, that might be obvious to more experienced users. Right after  loading a MD topology (.tpr) and trajectory (.xtc), I can play back the trajectory by scrolling through the frames. But when I close the file and reopen it again, the same is no longer possible. I checked and the file has a .MNsession file deposited in the same directory. Is there a way to keep the frames available when closing the file? Since I am not sure where the frames for the trajectory are deposited after loading, I'm also having trouble accessing them through geometry nodes and have so far operated solely by using the offset and subframe options in the Molecular nodes input interface for MD data.
All help is greatly appreciated! 

 (I'm using blender 4.2)

**Comment by BradyAJohnston** [comment]
It should be maintaining the connection to the frames after you re-open the file, if is losing connection then this is a bug & my apologies!

In the short term, if a trajectory has lost connection to it's underlying universe, the `Object` tab in the Molecular Nodes panel should prompt you to reload the trajectory. It should be default have the topology and trajectory filepaths previously used, and all you should need to do is hit `Reload Trajectory`. 

![CleanShot 2024-12-12 at 09 06 23](https://github.com/user-attachments/assets/014eb05f-dd04-4ee0-a84e-1e38121441cb)

If you could open an issue with any errors that are printing to console that would be very useful. On windows you can go `Window` -> `Show system console`, and on mac & linux if you start Blender from the command line then the errors will print to the console.

I am unsure what you are wanting to do with the final point. We no longer load everything into memory, so the frames aren't stored anywhere inside of Blender. When the timeline changes, then MDAnalysis under the hood fetches the new frame and updates the object's positions.

Because of this Geomtry Nodes doesn't have direct access to individual frames anymore, but you can use the `Bake` node or `Simulation Node` to cache some of the frames after playback and then get access to them, depending on what you are wanting to do.

**Reply by lnrdwsh1ngt0n** [reply]
Thank you for the response. 
Considering the problem, I figured out what was the problem (yet I don't understand why, but that's not really important). I accessed the files from two seperate PCs and even though they both have access to the server where the trajectories are stored, they only remained available when opening the blender file on the device where I had created it. I wouldnt really call that a bug. 
Thank you anyway for the advice, it helped me to understand the program a little better and Im sure eventually I'll be able to use that nevertheless.

Its pretty cool to get in touch with the developer himself! I want to take that chance to express my gratitude! MN is such a great tool and using it has been really fun! If there's a way to support the work, let me know.

**Reply by BradyAJohnston** [reply]
@lnrdwsh1ngt0n was the network drive named the same letter on both machines? If it is named the same in theory it should work, but if the drives are listed different or accessible through a different path then that'll break it. Something I haven't experimented much with but interested to hear more about your experience with it.
